src/users/users.service.ts

```diff
@@ -17,6 +17,14 @@ export class UsersService {
     createUserDto: CreateUserDto,
   ): Promise<DocumentType<UserSchema>> {
     const hashedPassword = await argon2.hash(createUserDto.password);
+    const existingUser = await this.userModel.findOne({
+      username: createUserDto.username,
+    });
+    if (existingUser) {
+      throw new UnauthorizedException({
+        message: 'Username is already taken',
+      });
+    }
```

Comment:
Unique is assigned when you create user it will throw error E1100?
https://stackoverflow.com/questions/24430220/e11000-duplicate-key-error-index-in-mongodb-mongoose,Please check from this.